pyhf/modifiers/combined_mods.py

```diff
@@ -258,7 +259,11 @@ def apply(self,pars):
         if not tensorlib.shape(shapefactor_indices)[0]:
             return
         shapefactors = tensorlib.gather(pars,shapefactor_indices)
-        results_shapefactor = shapefactor_mask * tensorlib.reshape(shapefactors,tensorlib.shape(shapefactors) + (1,1))
+
+        # needs to multiply across samples/mask value so (a,b) -> (a, 1, 1, b)
+        new_shape = list(tensorlib.shape(shapefactors))
+        new_shape[1:1] = [1,1]
+        results_shapefactor = shapefactor_mask * tensorlib.reshape(shapefactors, new_shape)
```

Comment:
this took some playing around, but for N shapefactors applied to M bins, we need to go from a shape of `shapefactor_mask.shape = (N, M)` to a `shapefactor_mask.shape = (N, 1, 1, M)` in order to multiply across correctly.